sdk/core/src/main/java/com/klaviyo/core/auth/KlaviyoAuthTokenManager.kt

```
@@ -309,6 +389,29 @@ internal class KlaviyoAuthTokenManager(
+    /**
+     * Iterates [refreshObservers] and invokes each with [jwt]. Best-effort: if an observer throws,
+     * the exception is logged at WARNING and the remaining observers are still called.
+     */
+    private fun notifyRefreshObservers(jwt: String) {
+        refreshObservers.forEach { observer ->
+            try {
+                observer(jwt)
+            } catch (e: CancellationException) {
+                // Structured-concurrency contract: CancellationException must never be swallowed.
+                throw e
+            } catch (e: Throwable) {
```

detekt (1.23.8)

[warning] The caught exception is too generic. Prefer catching specific exceptions to the case that is currently handled.

(detekt.exceptions.TooGenericExceptionCaught)